lib/endpoint-with-defaults.ts

```diff
@@ -0,0 +1,6 @@
+import merge = require('./merge')
+import parse = require('./parse')
+
+export = function endpointWithDefaults (defaults: typeof import("./defaults"), route: string, options: any) {
```

Comment:
can we move `exports = endpointWithDefaults` to the top of the file, in general? I’m usually interested in what a file is exporting the most and putting it separately above everything else helps with code readability, at least for me :)